library/std/src/os/linux/tests.rs

```diff
@@ -0,0 +1,28 @@
+use crate::fmt;
+use crate::net::test::next_test_ip4;
+use crate::net::{TcpListener, TcpStream};
+use crate::os::linux::net::TcpStreamExt;
+
+macro_rules! t {
+    ($e:expr) => {
+        match $e {
+            Ok(t) => t,
+            Err(e) => panic!("received error for `{}`: {}", stringify!($e), e),
+        }
+    };
+}
+
+#[test]
+#[cfg_attr(target_env = "sgx", target_os = "linux", ignore)]
```

Comment:
That cfg doesn't make sense, sgx will never show up as linux.

Instead the whole test should simply be cfg'ed to only run on target_os android or linux.

Reply:
Yeah about android, I don't know if this option is supported. The official (java) socket API only lists `TCP_NODELAY` as an option. Also don't know how to test this as I don't have an android device.

Reply:
The rust-on-android means native code, not java code. Since setting this just uses the `setsockopt` syscall it has been available on linux kernels since forever, so there shouldn't be a compatibility problem. At worst it'll return an error that the particular socket option isn't supported but that's covered by the `Result` return type.

Reply:
okay that's clear, ill change it